package/MDAnalysis/coordinates/H5MD.py

```diff
@@ -826,3 +826,301 @@ def __setstate__(self, state):
+            number of atoms in trajectory
+        n_frames: int
+            number of frames in trajectory
+        convert_units : bool (optional)
+            convert units from MDAnalysis to desired units
+        **kwargs : dict
+
+        """
+
+        self.filename = filename
+        if n_atoms == 0:
+            raise ValueError("H5MDWriter: no atoms in output trajectory")
+        self._driver = driver
+        self._comm = comm
+        self.n_atoms = n_atoms
+        self.n_frames = n_frames
+        self.units = units
+        self.compression = compression
+        self.convert_units = convert_units
+        self.data_blacklist = ['step', 'time', 'dt']
+
+        self.h5md_file = None
+        self.has_positions = kwargs.get('positions', False)
+        self.has_velocities = kwargs.get('velocities', False)
+        self.has_forces = kwargs.get('forces', False)
+        self._has = {'position': self.has_positions,
+                     'velocity': self.has_velocities,
+                     'force': self.has_forces}
+        self.chunks = kwargs.get('chunks', None)
+
+        # Pull out various keywords to store in 'h5md' group
+        self.author = kwargs.pop('author', 'N/A')
+        self.author_email = kwargs.pop('author_email', None)
+        self.creator = kwargs.pop('creator', 'MDAnalysis')
+        self.creator_version = kwargs.pop('creator_version', mda.__version__)
+
+    def _write_next_frame(self, ag):
+        """Write information associated with ``ag`` at current frame
+        into trajectory
+
+        Parameters
+        ----------
+        ag : AtomGroup or Universe
+
+        .. versionadded:: 2.0.0
+
+        """
+        try:
+            # Atomgroup?
+            ts = ag.ts
+            self.units = ag.universe.trajectory.units
+        except AttributeError:
+            try:
+                # Universe?
+                ts = ag.trajectory.ts
+                self.units = ag.trajectory.units
+            except AttributeError:
+                errmsg = "Input obj is neither an AtomGroup or Universe"
+                raise TypeError(errmsg) from None
+
+        if ts.n_atoms != self.n_atoms:
+            raise IOError("H5MDWriter: Timestep does not have"
+                          " the correct number of atoms")
+
+        # Opens H5MD file to write with H5PY library and initializes
+        # 'h5md' group and checks if unitcell is periodic
+        if self.h5md_file is None:
+            self._init_h5md(ts)
+
+        return self._write_next_timestep(ts)
+
+    def is_periodic(self, ts):
+        """Test if timestep ``ts`` contains a periodic box.
+
+        Parameters
+        ----------
+        ts : :class:`Timestep`
+             :class:`Timestep` instance containing coordinates to
+             be written to trajectory file
+
+        Returns
+        -------
+        bool
+            Return ``True`` if `ts` contains a valid simulation box
+        """
+        return np.all(ts.dimensions > 0)
```

Comment:
Just passing by (strangely enough was thinking of #2698 and saw this pop up), going by line 622, wouldn't a no-dimensions case set `ts.dimensions` to `None` (i.e. this should cause a `TypeError`)?